### PR TITLE
cleanup(config): improve `falco.yaml` config

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

Make it easier for adopters to understand and navigate Falco configs to achieve optimal deployment settings.

- re-arrange falco.yaml configs in logical categories
  - add an index for logical categories
  - move configs around without changing description content,
    solely add a uniform header to each config
  - indicate "Stable" or "Experimental" for most configs
    to indicate current stability or maturity
 - improve config descriptions for the basic config options

Was planning to touch up the remaining configs, but perhaps in a follow up PR depending on your preferences.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
cleanup(config): improve falco config
```

### CODE REVIEW

File: falco.yaml [hunk 1]
```diff
@@ -15,112 +15,363 @@
...
+# When using JSON output in Falco, you have the option to include the "tags"
+# field of the rules in the generated JSON output. The "tags" field provides
+# additional metadata associated with the rule. To reduce the logging volume,
+# if the tags associated with the rule are not needed for your use case or can
+# be added at a later stage, it is recommended to turn it off.
 json_include_tags_property: true
 
-# Send information logs to stderr and/or syslog Note these are *not* security
-# notification logs! These are just Falco lifecycle (and possibly error) logs.
+# Falco supports various output sinks, such as syslog, stdout, file, gRPC,
```
Comment (leogr) [inline falco.yaml:157]:
```suggestion
# Falco supports various output channels, such as syslog, stdout, file, gRPC,
```

We usually used the word "channel".

File: falco.yaml [hunk 2]
```diff
@@ -15,112 +15,363 @@
...
+# messages are directed. This flexibility allows seamless integration with your
+# preferred logging and alerting systems. Multiple outputs can be enabled
+# simultaneously.
+stdout_output:
+  enabled: true
+
+syslog_output:
+  enabled: true
+
+# When appending Falco alerts to a file, each new output will be added to a new
```
Comment (leogr) [inline falco.yaml:169]:
```suggestion
# When appending Falco alerts to a file, each new notification will be added to a new
```
?

Comment (leogr) [reply falco.yaml:169]: Or "alert".

File: falco.yaml [hunk 3]
```diff
@@ -152,314 +402,250 @@ syscall_event_drops:
...
   chunk_wait_us: 1000
   watch_freq_sec: 1
 
+# [Experimental] `load_plugins` and `plugins`
+#
+# Setting this list to empty ensures that the above plugins are *not*
+# loaded and enabled by default. If you want to use the above plugins,
+# set a meaningful init_config/open_params for the cloudtrail plugin
+# and then change this to:
+# load_plugins: [cloudtrail, json]
```
Comment (leogr) [inline falco.yaml:552]:
```suggestion
# Setting this list to empty ensures that the below plugins are *not*
# loaded and enabled by default. If you want to use the below plugins,
# set a meaningful init_config/open_params. For example, if you want to
# use the k8saudit plugin, ensure it is configured appropriately and then
# change this to:
# load_plugins: [k8saudit, json]
```

Comment (leogr) [reply falco.yaml:552]: `plugins` list is below now. Also, minor changes.

**Review by leogr — COMMENTED** [review]
I really like this! :star_struck: Thank you! :pray: 

I agree it's needed since we added a lot of options but never reorganized this file.

I've left some suggestions, mainly regarding the placement of the options. Apart from that, it seems already good for me!

**Comment by incertum** [comment]
@leogr thanks for the amazing suggestions, incorporated them all. Please let me know if we first want to stop here now and move further language adjustments and improvements to a new PR? Would recommend not overloading this PR as we have been moving around things a lot.

Loaded the yaml and printed all primary keys to ensure we didn't lose one and the ordering is correct, please also verify.

**Comment by incertum** [comment]
@leogr as discussed added all planned changes in this PR. From my perspective this PR can be considered complete.
Re `metadata_download` unsure if I got it right, would you mind helping me with this config?

Looking forward for the next round of feedback in order to polish everything.

File: falco.yaml [hunk 4]
```diff
@@ -152,525 +535,474 @@ syscall_event_drops:
...
+# Falco cloud orchestration systems integration #
+#################################################
+
+# [Stable] `metadata_download`
+# These parameters are currently specific to the `k8saudit` plugin.
+# 
+# When connected to an orchestrator like Kubernetes, Falco provides parameters
+# to configure the download behavior. Please note that support for Mesos is
+# deprecated. The following parameters can be used to customize the download
+# behavior:
```
Comment (leogr) [inline falco.yaml:921]:
```suggestion
# [Stable] `metadata_download`
When connected to an orchestrator, such as Kubernetes, Falco collects metadata, 
enhancing system call events with contextual data.
These parameters control the downloading process of this metadata.

Please note that support for Mesos is deprecated. So these currently apply only 
to the Kubernetes support when enabled using `-k` / `-K` command line flags.

The following parameters can be used to customize the download behavior:
```

These parameters are NOT related to the `k8saudit` plugin. They are actually sinsp settings that take effect when sinsp fetches k8s. for the metadata from the K8S API server to enrich syscall events. This feature is enabled when using `-K/-k` in Falco.
(in the past, those parameters applied to the sinsp Mesos client too, but now it's gone, so they only affect the sinsp K8S client).

Thus, `metadata_download` is a group of knobs that affects the ability to fetch metadata from the k8s API server, influencing Falco's userspace performance when processing syscalls.

So I've tried to correct the description. I just focused on the content, so feel free to change the style or adapt it as you prefer.

**Review by leogr — COMMENTED** [review]
I just left a comment for ` metadata_download`. Otherwise, SGTM.

And Melissa, thank you again for this PR :pray:

**Comment by incertum** [comment]
Addressed feedback, thank you!

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**